First draft of new wiki documentation (#27)
distributed-technologies/SCCP-documentation: README.md

> <img src="images/Energinet-logo.png" width="250" style="margin-bottom: 3%">

Reviewer: ```suggestion
![Energinet logo](images/Energinet-logo.png)
```

![Energinet logo](images/Energinet-logo.png)

# Overview

> The Secure Compute and Communication Platform (SCCP) project is an attempt at creating a modern, fast, secure, robust, and uniform platform for running a variety of application suites for running and monitoring the Danish energy grid.

Reviewer: ```suggestion
The Secure Compute and Communication Platform (SCCP) project is an attempt at creating a modern, fast, secure, robust, and uniform platform for running a variety of application suites for running and monitoring infrastructure.
```

The platform is made up of multiple components: 

### Hardware

> Hardware plays an essential part in the design of the platform. The platform rests on multiple servers that participate in a kubernetes cluster which provide redundancy. 

Reviewer: ```suggestion
Hardware plays an essential part in the design of the platform. The platform rests on multiple servers that participate in a Kubernetes cluster which provide redundancy. 
```

3. We would like to be able to monitor the system with itself
4. Infrastructure as code principles followed
5. We will make Helm releases of any service or application, so an upgrade to a newer version of the platform does not require pulls, but rather pointing to a newer version
6. The platform has to be highly configurable to anyone using it
7. We would like to use ArgoCD as our continuous delivery tool
8. Allow certain services to boot before others

These are just some of the many requirements there were for the platform and many more have come forward during development. 

### Yocto
After having completed a successfull PoC of the platform, we soon realized that the build method we were using was slow and did not have all the features we needed. We spent some time looking for other build tools and found the Yocto project. Yocto takes advantage of caching layers of the operating system, which means that if anything changes in a particular layer, only that layer has to be recompiled. This sped up OS compiling significantly and has allowed us to automatic builds. 

# Decision log
Any decisions made during meetings about SCCP that can be open to the public should be listed here.

> - 11.01.2022: A decision was made about looking into the possibility of packaging Yggdrasil and only requiring an environment repository to configure the platform.

Reviewer: Perhaps this should be split out into its own file?

Author: I have done this in the newest commit

> # Decision log
> Any decisions made during meetings about SCCP that can be open to the public should be listed here.
> 
> - 11.01.2022: A decision was made about looking into the possibility of packaging Yggdrasil and only requiring an environment repository to configure the platform.

Reviewer: https://xkcd.com/1179/

```suggestion
# Decision log
Any decisions made during meetings about SCCP that can be open to the public should be listed here.

- 2021.01.11: A decision was made about looking into the possibility of packaging Yggdrasil and only requiring an environment repository to configure the platform.
```
